src/components/light.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+export class Light extends React.Component {
+    onClick = () => {
+        this.props.handleClick(this.props.row, this.props.col);
```

Comment:
Since the parent component is passing `row`, `col`, `handleClick`, and `shouldBeOn`, it could probably get away with only passing the `handleClick` method, and passing the `on` state as a boolean instead:

```js
<Light
    handleClick={() => this.handleClick(row, col)}
    isOn={this.shouldBeOn(row, col)}
/>
```

Reply:
Really good solution! I thought for awhile on how to avoid passing in so many props but I could figure out how to work the handleClick function without passing in row and col, but of course using an arrow function to fix the scope makes perfect sense!